pkg/controller/keyvault/ingress_secret_provider_class.go

```diff
@@ -112,6 +112,9 @@ func (i *IngressSecretProviderClassReconciler) Reconcile(ctx context.Context, re
 	if ok {
 		logger.Info("reconciling secret provider class for ingress")
 		err = util.Upsert(ctx, i.client, spc)
+		if err != nil {
```

Comment:
these are hard to unit test because it's hard to simulate what causes upsert errors (primarily admission webhooks / policy webhooks) with the fake client.

I think it's okay not to unit test for now because this is a pretty straightforward way of using native k8s code. we are just using the k8s packages to do their intended purposes